extensions/vscode/src/multiStepInputs/newCredential.ts

```diff
@@ -455,28 +620,79 @@ export async function newCredential(
   // before completing the steps. This also serves as a type guard on
   // our state data vars down to the actual type desired
   if (
-    // have to add type guards here to eliminate the variability
+    // Common required fields for all authentication methods
     state.data.url === undefined ||
     isQuickPickItem(state.data.url) ||
-    state.data.apiKey === undefined ||
-    isQuickPickItem(state.data.apiKey) ||
-    state.data.snowflakeConnection === undefined ||
-    isQuickPickItem(state.data.snowflakeConnection) ||
     state.data.name === undefined ||
     isQuickPickItem(state.data.name)
   ) {
     return;
   }
 
+  // Check authentication method specific fields
+  if (isConnect(serverType)) {
+    if (state.data.authMethod === "token") {
+      // For token authentication, require token and privateKey
+      if (
+        state.data.token === undefined ||
+        isQuickPickItem(state.data.token) ||
+        state.data.privateKey === undefined ||
+        isQuickPickItem(state.data.privateKey)
+      ) {
+        return;
+      }
+    } else {
+      // For API key authentication, require apiKey
+      if (
+        state.data.apiKey === undefined ||
+        isQuickPickItem(state.data.apiKey)
+      ) {
+        return;
+      }
+    }
+  } else if (isSnowflake(serverType)) {
+    // For Snowflake, require snowflakeConnection
+    if (
+      state.data.snowflakeConnection === undefined ||
+      isQuickPickItem(state.data.snowflakeConnection)
+    ) {
+      return;
+    }
+  }
```

Comment:
Is this checking helpful here?

Reply:
_nods_ in the service of moving this PR forward, I'm going to defer that, especially because, as you say, this is getting reworked soon already.